cloud-pubsub/main.py

```diff
@@ -8,37 +8,43 @@
 import datetime
 import time
 
-from google.cloud import pubsub
-
-PUBSUB_TOPIC = 'echo'
-PUBSUB_SUBSCRIPTION = 'echo-read'
 # [START gke_pubsub_pull]
 # [START container_pubsub_pull]
+from google import auth
+from google.cloud import pubsub_v1
+
+
 def main():
     """Continuously pull messages from subsciption"""
-    client = pubsub.Client()
-    subscription = client.topic(PUBSUB_TOPIC).subscription(PUBSUB_SUBSCRIPTION)
-
-    print('Pulling messages from Pub/Sub subscription...')
-    while True:
-        with pubsub.subscription.AutoAck(subscription, max_messages=10) as ack:
-            for _, message in list(ack.items()):
-                print("[{0}] Received message: ID={1} Data={2}".format(
-                    datetime.datetime.now(),
-                    message.message_id,
-                    message.data))
-                process(message)
-
-
-def process(message):
-    """Process received message"""
-    print("[{0}] Processing: {1}".format(datetime.datetime.now(),
-                                         message.message_id))
-    time.sleep(3)
-    print("[{0}] Processed: {1}".format(datetime.datetime.now(),
-                                        message.message_id))
+
+    # read default project ID
+    _, project_id = auth.default()
+    subscription_id = 'echo-read'
```

Comment:
question: is hard-coding a subscription id to be `echo-read` required? i'm in favor of changing it to `${TOPIC_ID}-sub` to let users know it is a subscription paired with a specific topic, with a comment saying this subscription is attached to this topic

Reply:
Yes, it is. It follows the tutorial that instructs to create a fixed subscription name.